src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier randomColor = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Name[] names = Name.values();
+        int index = random.nextInt(names.length);
+        if (names[index].toString().equals("SQUARE")) {
+            return new Square(
+                     random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        }
+        if (names[index].toString().equals("RECTANGLE")) {
+            return new Rectangle(
+                          random.nextDouble(),
+                          random.nextDouble(),
+                          randomColor.getRandomColor(),
+                          names[index].toString()

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. Also, use the `name()` method instead of `toString()` for getting String representation of enum constants.